File: android/app/src/main/java/com/dbtutorialone/BiometricAuth.java
package com.dbtutorialone;

import androidx.annotation.NonNull;
import androidx.biometric.BiometricPrompt;
import androidx.fragment.app.FragmentActivity;

public class BiometricAuth {
    private FragmentActivity activity;

    public BiometricAuth(FragmentActivity activity) {
        this.activity = activity;
    }

    public void authenticate(BiometricCallback callback) {
        BiometricPrompt.PromptInfo promptInfo = new BiometricPrompt.PromptInfo.Builder()
                .setTitle(callback.getTitle())
                .setSubtitle(callback.getSubTitle())
                .setNegativeButtonText(callback.getCancel())
                .build();

        BiometricPrompt biometricPrompt = new BiometricPrompt(activity, new MainThreadExecutor(),
                new BiometricPrompt.AuthenticationCallback() {
                    @Override
                    public void onAuthenticationError(int errorCode, @NonNull CharSequence errString) {
                        super.onAuthenticationError(errorCode, errString);
                        callback.onError(errorCode, errString.toString());
                    }

                    @Override
                    public void onAuthenticationSucceeded(@NonNull BiometricPrompt.AuthenticationResult result) {
                        super.onAuthenticationSucceeded(result);
                        callback.onSuccess();
                    }

                    @Override
                    public void onAuthenticationFailed() {
                        super.onAuthenticationFailed();
                        callback.onAuthFailed();
                    }
                });

        biometricPrompt.authenticate(promptInfo);
    }

    public interface BiometricCallback {
        void onSuccess();

        void onError(int errorCode, String errorString);

        void onAuthFailed();

        String getTitle();

        String getSubTitle();

        String getCancel();
    }
}

File: android/app/src/main/java/com/dbtutorialone/BiometricAuth.kt
package com.dbtutorialone

import androidx.biometric.BiometricPrompt
import androidx.fragment.app.FragmentActivity


class BiometricAuth(private val activity: FragmentActivity) {

    fun authenticate(callback: BiometricCallback) {
        val promptInfo = BiometricPrompt.PromptInfo.Builder()
            .setTitle(callback.title)
            .setSubtitle("Authenticate using your biometric credentials")
            .setNegativeButtonText("Cancel")
            .build()

        val biometricPrompt = BiometricPrompt(activity, MainThreadExecutor(),
            object : BiometricPrompt.AuthenticationCallback() {

                override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
                    super.onAuthenticationSucceeded(result)
                    callback.onSuccess()
                }

                override fun onAuthenticationFailed() {
                    super.onAuthenticationFailed()
                    callback.onError()
                }

            })

        biometricPrompt.authenticate(promptInfo)
    }

    interface BiometricCallback {
        fun onSuccess()
        fun onError()
        val title:String
    }


}
